src/js/gutenberg-plugin.js

```diff
@@ -159,19 +248,81 @@ const ClassifAIGenerateTagsButton = () => {
 		postTypeLabel
 	);
 
-	return (
+	let updatedTaxQuery = Object.entries( taxQuery || {} ).reduce(
+		( accumulator, [ taxonomySlug, terms ] ) => {
+			accumulator[ taxonomySlug ] = terms;
+
+			return accumulator;
+		},
+		{}
+	);
+
+	if ( updatedTaxQuery.taxQuery ) {
+		updatedTaxQuery = updatedTaxQuery.taxQuery;
+	}
+
+	const modalData = (
 		<>
+			<TaxonomyControls
+				onChange={ ( newTaxQuery ) => {
+					setTaxQuery( newTaxQuery );
+				} }
+				query={ {
+					contentPostType: 'page',
```

Comment:
This is hardcoded to `page`, so won't work unless you're on a `page` post type. This should be dynamic